### PR TITLE
Rename CodeMap and FileMap to SourceMap and SourceFile.

### PR DESCRIPTION
**Pull request by ekse**
Those types were renamed in https://github.com/rust-lang/rust/pull/52953.

Note: nightly-rustc was not updated since the change was merged. We might to wait for that to happen before merging this PR to avoid having broken links in the meantime.

### CODE REVIEW

File: src/appendix/code-index.md [hunk 1]
```diff
@@ -7,20 +7,20 @@ compiler.
...
 `ParamEnv` | struct | Information about generic parameters or `Self`, useful for working with associated or generic items | [Parameter Environment] | [src/librustc/ty/mod.rs](https://doc.rust-lang.org/nightly/nightly-rustc/rustc/ty/struct.ParamEnv.html)
 `ParseSess` | struct | This struct contains information about a parsing session | [The parser] | [src/libsyntax/parse/mod.rs](https://doc.rust-lang.org/nightly/nightly-rustc/syntax/parse/struct.ParseSess.html)
 `Rib` | struct | Represents a single scope of names | [Name resolution] | [src/librustc_resolve/lib.rs](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_resolve/struct.Rib.html)
 `Session` | struct | The data associated with a compilation session | [The parser], [The Rustc Driver] | [src/librustc/session/mod.html](https://doc.rust-lang.org/nightly/nightly-rustc/rustc/session/struct.Session.html)
+`SourceFile` | struct | Part of the `SourceMap`. Maps AST nodes to their source code for a single source file | [The parser] | [src/libsyntax_pos/lib.rs](https://doc.rust-lang.org/nightly/nightly-rustc/syntax/source_map/struct.SourceFile.html)
```
Comment (mark-i-m) [inline src/appendix/code-index.md]: Perhaps add a note that this was previously called the `FileMap`. That way people stumbling on old issues will have a point of reference.

Same for `SourceMap`

Comment (ekse) [reply src/appendix/code-index.md]: done!

**Comment by ekse** [comment]
r=mark-i-m

**Comment by ekse** [comment]
r=@mark-i-m

**Pull request merged**